ColorzCore/Parser/EAParser.cs

```diff
@@ -79,6 +82,9 @@ public EAParser(Dictionary<string, IList<Raw>> raws, Log log, DirectiveHandler d
             Definitions = new Dictionary<string, Definition>();
             Inclusion = ImmutableStack<bool>.Nil;
             this.directiveHandler = directiveHandler;
+
+            PooledLines = new List<List<Token>>();

```

Comment:
Error or warning upon completion of parsing if the pool is not emptied at the end.

Reply:
Alternatively, thoughts on associating pooled lines to scope, and thus erroring if there are still pooled lines at the end of a scope?

Or do you think pooled lines should be able to escape to outer scopes?

Reply:
I don't like this. Putting extra data somewhere without the user explicitly asking for it sounds like a bad idea.

Reply:
I think you misunderstand me. I mean if I have
```
{
AddToPool(blah)
}
#pool
```

Should this error? Arguably, pooled data should be required only within its scope, so lack of a #pool in the scope may be an error, so we may want to enforce pooling on a scope level... was my thought

Reply:
:thinking: I think pooling outside of scope that way is fine, I think it is only problematic if the `#pool` were to be inside a child scope and not parent scope... But even then maybe it isn't idk

```
{
AddToPool(blah)
}
#pool // ok
```

```
AddToPool(blah)
{
#pool // maybe not ok?
}
```

(but yes I did misread/read too quickly and thought doing silent stuff instead of err sorry).

Reply:
Wait I think I see a (technical) issue now: you wouldn't be able to access scope local labels referenced within the AddToPool expression if the corresponding `#pool` was outside... ugh

Reply:
Let's associate pooled lines with the scope they're used in, then, and error at the end of a scope if there are undumped pooled lines.